pandas/core/arrays/timedeltas.py

```diff
@@ -997,6 +989,30 @@ def objects_to_td64ns(data, unit="ns", errors="raise"):
     return result.view('timedelta64[ns]')
 
 
+def _validate_td64_dtype(dtype):
+    try:
+        if dtype == np.dtype("timedelta64"):
+            dtype = _TD_DTYPE
+            msg = textwrap.dedent("""\
+                Passing in 'timedelta' dtype with no precision is deprecated
+                and will raise in a future version. Please pass in
+                'timedelta64[ns]' instead.""")
+            warnings.warn(msg, FutureWarning, stacklevel=4)
+    except TypeError:
+        # extension dtype
+        pass
+
+    try:
+        dtype_mismatch = dtype != _TD_DTYPE
+    except TypeError:
+        raise ValueError(_BAD_DTYPE.format(dtype=dtype))
```

Comment:
Note the change from TypeError to ValueError, when compared with https://github.com/pandas-dev/pandas/pull/24806/files#diff-57368e94ebd4c37c96935ab9862e3bc7L169. ValueError is more consistent with DatetimeArray.